WcaOnRails/app/models/user.rb

```diff
@@ -331,6 +331,10 @@ def avatar_requires_wca_id
     end
   end
 
+  def send_on_create_confirmation_instructions
```

Comment:
I think it would be good to have a test that confirms that this method actually gets called when a user registers. I think that would look like a test in `user_spec` that mocks out `NewRegistrationMailer.send_registration_mail` and asserts that it is called with this new user.

Reply:
Fixed tests and mail text!

I also created all necessary files for the new email address change confirmation mail as requested in https://github.com/thewca/worldcubeassociation.org/issues/1814. 

Screenshot:
![new-mail-confirmation](https://user-images.githubusercontent.com/18479675/42052431-1a64eb4e-7b0f-11e8-9cb8-34f2b19add2b.png)